include/libtorrent/aux_/torrent_peer.hpp

```diff
@@ -264,16 +264,14 @@ namespace libtorrent::aux {
 
 		bool operator()(torrent_peer const* lhs, torrent_peer const* rhs) const
 		{
-#if TORRENT_USE_I2P || TORRENT_USE_RTC
-			if (	true
 #if TORRENT_USE_I2P
-					&& rhs->is_i2p_addr == lhs->is_i2p_addr
-#endif
-#if TORRENT_USE_RTC
-					&& rhs->is_rtc_addr == lhs->is_rtc_addr
-#endif
-			)
-				return lhs->dest().compare(rhs->dest()) < 0;
+			if (lhs->is_i2p_addr || rhs->is_i2p_addr)
+			{
+				if(lhs->is_i2p_addr && rhs->is_i2p_addr)
+					return lhs->dest().compare(rhs->dest()) < 0;
+				else
+					return lhs->is_i2p_addr;
+			}
```

Comment:
is this restoring this to its original form?

Reply:
It is now. It would break if there is a mix a I2P peers and normal peers for a torrent, but it seems that's never supposed to happen.